### ISSUE TITLE
redis_cluster_load crash

### ISSUE THREAD

**Issue by yatsukhnenko** [phpredis/phpredis#867]
There are a lot of calls `estrdup(INI_STR(""redis.clusters.*"))` without checking INI params existence in `redis_cluster_load`
```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff5493eda in strlen () from /lib64/libc.so.6
(gdb) bt
#0  0x00007ffff5493eda in strlen () from /lib64/libc.so.6
#1  0x00000000007a74e8 in _estrdup ()
#2  0x00007fffec35875a in redis_cluster_load () from /usr/lib64/php5.5/lib/extensions/debug-zts-20121212/redis.so
#3  0x00007fffec358a30 in zim_RedisCluster___construct () from /usr/lib64/php5.5/lib/extensions/debug-zts-20121212/redis.so
#4  0x0000000000838690 in ?? ()
#5  0x0000000000839128 in ?? ()
#6  0x00000000008379bd in execute_ex ()
#7  0x0000000000837aa6 in zend_execute ()
#8  0x00000000007d8ab6 in zend_eval_stringl ()
#9  0x00000000007d8ea8 in zend_eval_stringl_ex ()
#10 0x00000000007d8f6f in zend_eval_string_ex ()
#11 0x00000000008b91ca in ?? ()
#12 0x00000000008ba624 in main ()
```

**Comment by yatsukhnenko** [comment]
ping @michael-grunder

**Comment by michael-grunder** [comment]
Looks good to me.  I'm planning on going through several of these issues this afternoon.

Once I pull it down I'll port it into the php7 branch as well.

Cheers
Mike

**Comment by michael-grunder** [comment]
Yep, feel free :smiley:

Just make sure it still compiles in php7, etc.  The php7 API has a few gotchas, but this might work identically (not sure).

**Comment by yatsukhnenko** [comment]
Merged  #869 and #878
